### ISSUE TITLE
HIGH: [Reliability] Multiple calls to `GetNewerActions` on report open

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#39674]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.60-7
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1712180926567259

## Action Performed:
1. Sign in to NewDot
2. Open any report twice and don't scroll anywhere

## Expected Result:
Shouldn't call `GetNewerActions` multiple times

## Actual Result:
`GetNewerActions` called  5times, despite not scrolling anywhere.  Also, all 5 calls seem identical

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![image (10)](https://github.com/Expensify/App/assets/38435837/816cb143-99ee-4886-9dc8-ccca4d6d5aa7)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by twisterdotcom** [comment]
Asking a clarifying Q: https://expensify.slack.com/archives/C049HHMV9SM/p1712318653409559?thread_ts=1712180926.567259&cid=C049HHMV9SM

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by twisterdotcom** [comment]
https://github.com/Expensify/App/assets/9133401/ad6495ad-0e92-4da3-ab9e-734fec7a5519

**Comment by twisterdotcom** [comment]
I still can't reproduce this. Nobody else has reported it recently.

**Comment by twisterdotcom** [comment]
https://github.com/Expensify/App/assets/9133401/d319aa5d-38f9-4e9d-972f-dbdd55aedae0

So, an example of a single report loading `GetNewerActions` twice is https://staging.new.expensify.com/r/8036391930866711

When I load it, there are two `GetNewerActions` calls:

First:

```
jsonCode : 200
onyxData : [{onyxMethod: "merge", key: "reportActions_6028698164025499", value: {,…}}]
	0 : {onyxMethod: "merge", key: "reportActions_6028698164025499", value: {,…}}
		key : "reportActions_6028698164025499"
		onyxMethod : "merge"
	value : {,…}
		3459936395858833627 : {person: [{type: "TEXT", style: "strong", text: "CFO expensivepie.com"}], actorAccountID: 14459901,…}
			actionName : "REPORTPREVIEW"
			actorAccountID : 14459901
			actorEmail : "cfo@expensivepie.com"
			automatic : false
			avatar : "https://d1wpcgnaa73g0y.cloudfront.net/f427ffa485716562c8a975c8805a447b5684b740_128.jpeg"
			childCommenterCount : 1
			childLastActorAccountID : 8698376
			childLastReceiptTransactionIDs : "1204108464410787705"
			childLastVisibleActionCreated : "2024-03-14 21:38:55.951"
			childMoneyRequestCount : 1
			childOldestFourAccountIDs : "11665625"
			childRecentReceiptTransactionIDs : {1204108464410787705: "2024-03-14 21:37:29"}
				1204108464410787705 : "2024-03-14 21:37:29"
			childReportID : 1783551381563554
			childReportNotificationPreference : "hidden"
			childStateNum : 1
			childStatusNum : 1
			childType : "iou"
			childVisibleActionCount : 1
			created : "2024-03-14 21:37:29.911"
			lastModified : "2024-03-14 21:37:29.911"
			message : [,…]
				0 : {type: "COMMENT", html: "cfo@expensivepie.com owes A$0.00", text: "cfo@expensivepie.com owes A$0.00",…}
					deleted : ""
					html : "cfo@expensivepie.com owes A$0.00"
					isDeletedParentAction : false
					isEdited : false
					reactions : []
					text : "cfo@expensivepie.com owes A$0.00"
					type : "COMMENT"
					whisperedTo : []
				originalMessage : {lastModified: "2024-03-14 21:37:29.911", linkedReportID: "1783551381563554",…}
					lastModified : "2024-03-14 21:37:29.911"
					linkedReportID : "1783551381563554"
					originalActionAccountID : 14459901
				person : [{type: "TEXT", style: "strong", text: "CFO expensivepie.com"}]
					0 : {type: "TEXT", style: "strong", text: "CFO expensivepie.com"}
						style : "strong"
						text : "CFO expensivepie.com"
						type : "TEXT"
					previousReportActionID : "90093531819613915"
					reportActionID : "3459936395858833627"
					reportActionTimestamp : 1710452249911
					shouldShow : true
					timestamp : 1710452249
					whisperedToAccountIDs : []
	requestID : "872b3661fb7e63ec-LHR"
```

Second:
```
jsonCode : 200
onyxData : [{onyxMethod: "merge", key: "reportActions_8036391930866711", value: {,…}}]
	0 : {onyxMethod: "merge", key: "reportActions_8036391930866711", value: {,…}}
		key : "reportActions_8036391930866711"
		onyxMethod : "merge"
			value : {,…}
				2397210080192955362 : {person: [{type: "TEXT", style: "strong", text: "Admin Pie User"}], actorAccountID: 8698376,…}
					actionName : "REPORTPREVIEW"
					actorAccountID : 8698376
					actorEmail : "pecan@expensivepie.com"
					automatic : false
					avatar : "https://d1wpcgnaa73g0y.cloudfront.net/f7493165a790b796c92bd0337f8cf6b9bb74c669_128.jpeg"
					childLastActorAccountID : 8698376
					childMoneyRequestCount : 3
					childReportID : 438513267536357
					childReportNotificationPreference : "hidden"
					childType : "expense"
					created : "2024-03-16 13:30:42.420"
					lastModified : "2024-03-16 13:30:42.420"
					message : [{type: "COMMENT", html: "Xero owes $132.30", text: "Xero owes $132.30", isEdited: false,…}]
						0 : {type: "COMMENT", html: "Xero owes $132.30", text: "Xero owes $132.30", isEdited: false,…}
							deleted : ""
							html : "Xero owes $132.30"
							isDeletedParentAction : false
							isEdited : false
							reactions : []
							text : "Xero owes $132.30"
							type : "COMMENT"
							whisperedTo : []
						originalMessage : {lastModified: "2024-03-16 13:30:42.420", linkedReportID: "438513267536357"}
							lastModified : "2024-03-16 13:30:42.420"
							linkedReportID : "438513267536357"
						person : [{type: "TEXT", style: "strong", text: "Admin Pie User"}]
							0 : {type: "TEXT", style: "strong", text: "Admin Pie User"}
								style : "strong"
								text : "Admin Pie User"
								type : "TEXT"
						previousReportActionID : "5789708545474438065"
						reportActionID : "2397210080192955362"
						reportActionTimestamp : 1710595842420
						shouldShow : true
						timestamp : 1710595842
						whisperedToAccountIDs : []
	requestID : "872b3661fb7d63ec-LHR"
```

Maybe this is normal, maybe not?

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
HIGH: [Needs Reproduction] [Reliability] Multiple calls to GetNewerActions on report open
### What is the root cause of that problem?
When we navigate from A report to B the `loadNewerChats` will be called for both pages
### What changes do you think we should make in order to solve the problem?
Add `|| !isFocused`  to [this](https://github.com/Expensify/App/blob/7d19f827dcee2c9ffb69fdef5a5055ffcbf39b56/src/pages/home/report/ReportActionsView.tsx#L347C13-L347C177) condition. All

### What alternative solutions did you explore? (Optional)

**Comment by s77rt** [comment]
@shahinyan11 Thanks for the proposal. Your RCA is not complete. Can you explain why is the `loadNewerChats` function being called in the first place for the first report (report A)?

PS: I think that's a different bug than what's reported, I'm suspecting `GetNewerActions` is getting called for the **same** report multiple times, I didn't experience this much but got 3 total calls (one for the report A and two for report B)

<img width="744" alt="Screenshot 2024-04-11 at 9 52 48 PM" src="https://github.com/Expensify/App/assets/16493223/d890fdb2-1b9c-470a-8b4c-64503b81ddd3">

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/39674#issuecomment-2049803731). Added solution for two bugs.

**Comment by s77rt** [comment]
@shahinyan11 Thanks for the update. The root cause is still not correct. The call sequence is as follow: `loadNewerChats -> handleReportActionPagination -> fetchNewerAction` and in `fetchNewerAction` we have an early return checking for `isLoadingNewerReportActions`
https://github.com/Expensify/App/blob/b59ae94dbbfac2d3b5da8df5c4dc6fbdb7785c6b/src/pages/home/report/ReportActionsView.tsx#L193-L197


Thus not checking for `isLoadingNewerReportActions` in `loadNewerChats` is not the culprit. On the other hand I'm seeing unguarded calls to `getNewerActions` in `loadOlderChats` which could be the culprit. Let's investigate more on that

**Comment by shahinyan11** [comment]
@s77rt In our case the `getNewerActions` never called from `loadOlderChats` function.  My root cause may not be entirely correct, but you can test my solution and see if it fixes the error.

**Comment by s77rt** [comment]
@shahinyan11 Let's focus on narrowing down the root cause first

**Comment by shahinyan11** [comment]
@s77rt Ok I will work on it

**Comment by shahinyan11** [comment]
@s77rt With the latest version, I can't reproduce the case where the same `GetNewerActions` request is sent multiple times. The only remaining case is when `GetNewActions` is called from a previous report.

**Comment by s77rt** [comment]
@shahinyan11 Was not able to reproduce that bug either. And for the case where `GetNewActions` is called from the previous report, this is not reproducible all the time. The root cause for this one is still not clear


https://github.com/Expensify/App/assets/16493223/20561048-ffe0-4a1c-83af-1b5c0d27f6ee

**Comment by shahinyan11** [comment]
@s77rt  The `GetNewActions` request will not be called for the previous report if there are more than 23 `reportActions` in it.  [Here](https://github.com/Expensify/App/blob/c5ea7161015ff368883f92dea8d91bf0a7d498ff/src/pages/home/report/ReportActionsView.tsx#L331C1-L333C155) we prevent action if the `isLoadingOlderReportsFirstNeeded` is `true`

**Comment by s77rt** [comment]
@shahinyan11 `loadNewerChats` shouldn't be called in the first place. Re-rendering `ReportActionsList` on its own is not the culprit


https://github.com/Expensify/App/assets/16493223/d402fbf9-ecf9-401e-8096-c654395db780

**Comment by quinthar** [comment]
So I'm a bit confused, where does this stand?  What are the next steps, and who is doing them?

**Comment by twisterdotcom** [comment]
This is following the normal BugZero process, no changes to that. We're not yet understanding the root cause and @s77rt and @shahinyan11 are discussing it. It remains open to other External proposals.

**Comment by shahinyan11** [comment]
@s77rt `onStartReached` will be called for the previous report even if I manually prevent the `MVCPFlatList` component from re-rendering. I think it comes from the `react-native-web` package because I noticed `onStartReached` in the patch files  and I also  tested it on Native platforms and `onStartReached` is not called for the previous report after navigating to the new one

**Comment by s77rt** [comment]
@shahinyan11 Can you elaborate on this? Is the bug caused by the `react-native-web` patch?

**Comment by shahinyan11** [comment]
> @shahinyan11 Can you elaborate on this? Is the bug caused by the `react-native-web` patch?

Yes, this is the result of my research about calling `loadNewerChats`.

**Comment by s77rt** [comment]
@shahinyan11 Did you explore any possible solutions here? 

(also try revert the patch to be sure it's the culprit)

**Comment by twisterdotcom** [comment]
For opening a single report, why do I need to do all of these actions. Two `OpenReport`, two `GetNewerActions`, two `SendPerformanceTiming`. I guess most of it could be resolved by removing the second `GetNewerActions` call here, but I was surprised to see us call OpenReport again as well, and I've got to imagine they're linked.

<img width="321" alt="image" src="https://github.com/Expensify/App/assets/9133401/8bc383b4-e079-4139-a045-db36db3fc607">

**Comment by shahinyan11** [comment]
> (also try revert the patch to be sure it's the culprit)

I won't say it's because of the patch. Аrom the patch file I just realized that the package handles `onStartReached`. 
Tried deleting the patch file, but the behavior did not change.

**Comment by s77rt** [comment]
@shahinyan11 Please update your proposal and tag me once you get a clear understanding on the root cause.

PS: Keep in mind that we probably still have 2 bugs here, the GetNewerActions calls on same report (seems hard to reproduce) and the GetNewerActions call on the previous report

**Comment by shahinyan11** [comment]
@twisterdotcom @s77rt Would you mind increase the compensation here as this is not an easy subject and takes a lot of time to learn RCA.

**Comment by twisterdotcom** [comment]
Sure, we can make this $500, it's reliability. As long as we get to the root cause.

**Comment by quinthar** [comment]
Hi all, anybody able to take this on?  It's an important one!

**Comment by twisterdotcom** [comment]
Also discussed internally here: https://expensify.slack.com/archives/C01SKUP7QR0/p1713818410907859

Still would love to get some external proposals. @shahinyan11 any more thoughts here?

**Comment by shahinyan11** [comment]
@twisterdotcom If `onStartReached` should not be called on blur screen then it is a bug from `react-native-web` but currently I do not have any idea how to fix that in package. But if it is expected behavior then I already proposed solution

**Comment by twisterdotcom** [comment]
@s77rt what do you think? It's kind of over my head I'm afraid.

**Comment by s77rt** [comment]
Reminder: We have two bugs:
 1. On opening a report we make multiple calls to GetNewerActions (onStartReached)
 2. On unmounting a report we make an extra call to GetNewerActions (onStartReached)


I tried to isolate the problem by creating a snack https://snack.expo.dev/IDlFHIFq6DZP2MIDaqs-T and I can tell that the first problem is in react-native and the second is from our side.

Results:
- Native
    - On first render: onStartReached is called multiple times 🐛 
    - On unmount: onStartReached is not called ✅ 
- Web:
    - On first render: onStartReached is called multiple times 🐛 
    - On unmount: onStartReached is not called ✅ 


Regarding the first bug, upon further testing I concluded that onStartReached will be called upto n times where `n = Math.ceil(data.length / initialNumToRender)`. The more report actions in a chat the more the buggy behaviour is experienced.




https://github.com/Expensify/App/assets/16493223/2387505c-13cb-4ddd-a2d8-92b515b11489

**Comment by s77rt** [comment]
From ^ We have to fix the first bug at react-native level and second bug at /App level.

**Comment by shahinyan11** [comment]
@s77rt Thanks for providing useful information. But we make an extra call to GetNewerActions (onStartReached) not when unmounting, but when the screen is blurred. And this behavior only happens on `Web`
I also created a snack (https://snack.expo.dev/@shahinyan11/flatlist-simple?platform=web with navigation. And you can see that the second problem also is not from our side.

https://github.com/Expensify/App/assets/46921547/708b59b8-e877-4f9b-af6f-4aa85f66eb29

**Comment by s77rt** [comment]
@shahinyan11 You are right thanks for checking that! Both bugs are upstream

**Comment by s77rt** [comment]
Bugs summary:

- RN (+RNW): On mount onStartReached is called multiple times, should be called once (or none?)
- RNW: On focus/blur onStartReached is called once, should not be called

**Comment by twisterdotcom** [comment]
Wow, amazing work here both. Okay, so which repos do we need fixes in? Just react-native-web? We typically pay higher bounties for upstream fixes and I'm willing to commit to that here.

**Comment by s77rt** [comment]
@shahinyan11 Upstream bugs require upstream PRs. Handling any /App would have to be a workaround

**Comment by twisterdotcom** [comment]
I'm open to a workaround for now if @s77rt agrees it's not going to be too detrimental, as upstream PRs often take ages to merge and this is something we think might be causing a fair few issues.

**Comment by s77rt** [comment]
Finding the root cause on RNW may be easier. We can start by checking the number of calls to `_maybeCallOnEdgeReached` and find in which case it's called more often i.e. on blur which function calls this one.

**Comment by s77rt** [comment]
@twisterdotcom For the RNW bug (the one extra call) the workaround is pretty straightforward which is to ignore `onStartReached` callbacks if the screen is not focused. For the RN bug (the many extra calls) no clear workaround yet

**Comment by twisterdotcom** [comment]
@janicduplessis who commented in Slack, would love to get you involved here. I'm leaning towards maybe just having @shahinyan11 handle the RNW workaround, and then assigning you to a separate issue to handle these in RN and RNW natively - does that sound alright?

**Comment by janicduplessis** [comment]
@twisterdotcom Sounds good, I can handle the RN part, I am pretty familiar with that code.

**Comment by s77rt** [comment]
@shahinyan11 Can you please update your proposal to fix the RNW bug (should fix both extra calls on focus and on blur)

**Comment by shahinyan11** [comment]
@s77rt would you mind using throttling to fix extra calls on focus ?

**Comment by janicduplessis** [comment]
What do you think about merging something like this so when not using comment linking we just never try to fetch newer messages as there will never be any. I think this can be a good improvement to this for now until we can have a better integration with the backend as discussed [here](https://expensify.slack.com/archives/C05LX9D6E07/p1714022514340709?thread_ts=1713958592.192089&cid=C05LX9D6E07).

https://github.com/Expensify/App/pull/41053

**Comment by janicduplessis** [comment]
I also think I found a good solution to help avoid invalid calls to onStartReached in react-native so that will help for the case where we are actually using comment linking and GetNewerActions should get called, but only once.

**Comment by shahinyan11** [comment]
@s77rt Can you help reproduce multiple call during focus?

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/39674#issuecomment-2049803731). Added solution which will fix two bugs.

**Comment by twisterdotcom** [comment]
@s77rt can we assign @shahinyan11?

**Comment by janicduplessis** [comment]
@shahinyan11 I’m pretty sure comment linking relies on the fact that onStartReached is called initially. Would you be able to test this to make sure your solution doesn’t break this?

I think it could be an ok temporary solution, but we should still implement other improvements mentioned here.

**Comment by shahinyan11** [comment]
> @shahinyan11 I’m pretty sure comment linking relies on the fact that onStartReached is called initially. Would you be able to test this to make sure your solution doesn’t break this?

I just added a new alternative solution. But can you explain what you mean by **comment linking?** . Could you describe an example

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/39674#issuecomment-2049803731). Added alternative solution

**Comment by janicduplessis** [comment]
@shahinyan11 If you take a chat with many messages you can right click on a message in the middle and Copy Link. Then in another chat you can link to that message. This will use GetNewerActions and onStartReached to load the newer messages so we have to make sure this still works properly.

<img width="519" alt="image" src="https://github.com/Expensify/App/assets/2677334/5b0519a3-02ed-42c9-b8ef-9c5f20e6931f">

**Comment by shahinyan11** [comment]
@janicduplessis Yes you are correct. My main solution will break it so I updated it to allow `onStartReached`  to be called once

**Comment by s77rt** [comment]
@shahinyan11 On focus the callback is still executed https://snack.expo.dev/aZqOHa5m9m1AoGHGqAhoN?platform=web Also why you are adding `e.distanceFromStart === 0` check?

**Comment by shahinyan11** [comment]
@s77rt It is necessary that the callback be executed once during focus. See this https://github.com/Expensify/App/issues/39674#issuecomment-2079352605 .
`distanceFromStart` returns the scroll distance that was scrolled before reaching the start and it always `0` on init

**Comment by quinthar** [comment]
What's the next step and ETA on this?

**Comment by janicduplessis** [comment]
I think there are a few things we can do to improve this. This issue has 2 main causes:

1. `onStartReached` is called more than it should. Ideally this would be fixed upstream.
2. We do not have a good way to know if calling GetNewerActions is needed. Ideally this would be returned by the API when opening the report so even if `onStartReached is called we don't fetch new reports.

For 1. I am working [on a PR](https://github.com/facebook/react-native/pull/44287) and have a draft implementation, working on testing it more and then it will need to be reviewed by meta engineers.

For 2. I discussed this with @roryabraham and he will tackle this task when he can, I will create an issue to track this.

In the meantime I suggest merging some sort of workaround as proposed here. I think the simplest and safest option would be to just never call GetNewerActions when we are not in a comment linking report as implemented [here](https://github.com/Expensify/App/pull/41053).

**Comment by s77rt** [comment]
@shahinyan11 

> It is necessary that the callback be executed once during focus

Let's limit that only for comment linking. In other cases calling `GetNewerActions` on focus is redundant 

> distanceFromStart returns the scroll distance that was scrolled before reaching the start and it always 0 on init

This didn't answer my question. I meant what's the purpose of this condition or what would go wrong if we don't add it

**Comment by s77rt** [comment]
@janicduplessis Let's go with the RN patch + disabling GetNewerActions for non-comment-linked reports

**Comment by s77rt** [comment]
**Current situation:**

|                             Bug                             |     RCA    | Fixed in Regular reports? | Fixed in Comment-Linked reports? |
|:-----------------------------------------------------------:|:----------:|:-------------------------:|:--------------------------------:|
| `GetNewerActions` called multiple times on rendering report | Bug in RN  | No 🐛                      | No 🐛                             |
| `GetNewerActions` called 1 time on focusing/blurring report | Bug in RNW | No 🐛                      | No 🐛                             |



**After @janicduplessis's solution:**

|                             Bug                             |     RCA    | Fixed in Regular reports? | Fixed in Comment-Linked reports? |
|:-----------------------------------------------------------:|:----------:|:-------------------------:|:--------------------------------:|
| `GetNewerActions` called multiple times on rendering report | Bug in RN  | Yes ✅                     | Yes ✅                            |
| `GetNewerActions` called 1 time on focusing/blurring report | Bug in RNW | Yes ✅                     | No 🐛                             |

**Comment by s77rt** [comment]
For the remaining unresolved case, it requires a react-native-web upstream PR. From /App side the safest we can do is to prevent the call on blurring the report (partly fixing the issue, the call would still occur on focusing the report)

**Comment by s77rt** [comment]
@janicduplessis Can you please raise a PR for both the patch and the suggested solution?

**Comment by janicduplessis** [comment]
First PR is up, will do the patch one later today or tomorrow!

**Comment by s77rt** [comment]
🎀 👀 🎀

**Comment by shahinyan11** [comment]
> This didn't answer my question. I meant what's the purpose of this condition or what would go wrong if we don't add it

@s77rt if `distanceFromStart` is `0` this means that cllback is called without the user scrolling ( i.e. on rendering ). if we don't add to the callback will only be called the first time on user scroll.

**Comment by shahinyan11** [comment]
Now what should I do next?

**Comment by roryabraham** [comment]
> I have https://github.com/facebook/react-native/pull/44287 ready, need to be reviewed by meta engineers. We could make a patch for it to use it now.

Thanks for the upstream PR @janicduplessis - feel free to open a PR with the patch.

**Comment by janicduplessis** [comment]
Yes, planning to do that today!

**Comment by janicduplessis** [comment]
@s77rt @roryabraham Here's the PR for the patch https://github.com/Expensify/App/pull/41189

**Comment by s77rt** [comment]
Looks like the RN patch also fixed the RNW bug 🎉

**Comment by janicduplessis** [comment]
I think we are good for this issue. Yes please pay out @shahinyan11 also for his help and work on this.

**Comment by shahinyan11** [comment]
@janicduplessis @janicduplessis Thank you for appreciating my work

**Comment by twisterdotcom** [comment]
Payment Summary (to be paid after 7 days):

- @shahinyan11 paid $250 [here](https://www.upwork.com/nx/wm/offer/102086769)
- @s77rt paid $500 [here](https://www.upwork.com/nx/wm/offer/102086809)
- @janicduplessis does not require payment.

**Comment by s77rt** [comment]
Can we close this? i think the next steps will be handled in https://github.com/Expensify/App/issues/41153

**Comment by francoisl** [comment]
@twisterdotcom did you get a chance to issue the payments you had summarized in [this comment](https://github.com/Expensify/App/issues/39674#issuecomment-2083610700)?

**Comment by twisterdotcom** [comment]
Yes, we can close.
